### PR TITLE
CHK-439: Add Modal and SelectedCard components

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?

Adds a new `Modal` and `SelectedCard` components to be used by several apps inside Checkout IO.

#### How should this be manually tested?

[Workspace](https://lucas--checkoutio.myvtex.com/cart/add?sku=289).

You can see these new components in the payment step of Checkout. The modal has a different animation on mobile than on desktop, which is a bottom-up slide instead of a fade-in animation.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

Desktop:

gif1

Mobile:

gif2

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

Depends on vtex/styleguide#1346.

### CODE REVIEW

**Review by jeffersontuc — APPROVED** [review]
Noice!

**Comment by wdsrocha** [comment]
question (ui): When opening the modal, the focus goes to the "close" button, creating a border around it (even on mobile). Is that on purpose?

**Comment by lucasecdb** [comment]
> question (ui): When opening the modal, the focus goes to the "close" button, creating a border around it (even on mobile). Is that on purpose?

yes, the close button is the first focusable element inside the modal, and this is required for a11y

**Comment by lucasecdb** [comment]
I'm working on a bug in the mobile animation right now, I'll probably need to request your reviews again later :-/

File: react/hooks/useFadeEffect.ts
Comment (wdsrocha) [inline react/hooks/useFadeEffect.ts]: question: what does "raf" means?

Comment (lucasecdb) [reply react/hooks/useFadeEffect.ts]: short for "requestAnimationFrame" 😅

Comment (wdsrocha) [reply react/hooks/useFadeEffect.ts]: wow, could you explicit this in a comment, please? Because even after some googling, I didn't found a relevant result lol

Comment (lucasecdb) [reply react/hooks/useFadeEffect.ts]: do you think typing it as the following could help?

```ts
const rafIdRef = useRef<ReturnType<typeof requestAnimationFrame> | null>(null)
```

Comment (wdsrocha) [reply react/hooks/useFadeEffect.ts]: Looks even better to me!

Comment (lucasecdb) [reply react/hooks/useFadeEffect.ts]: updated in 908bbf6 :-)

**Review by wdsrocha — APPROVED** [review]
🔝

**Comment by lucasecdb** [comment]
the docs bot isn't commenting, I'll temporarily remove the required from the check to merge this PR and will add it again after it's merged

**Pull request merged**
